app/panel/components/Panel.jsx

```diff
@@ -251,6 +273,23 @@ class Panel extends React.Component {
 		);
 	}
 
+	_renderInsightsPromoModal = () => {
+		const { account, isTimeForInsightsPromo } = this.props;
+		const { insightsPromoModalShown } = this.state;
+
+		if (insightsPromoModalShown || !isTimeForInsightsPromo) return null;
+		if (account && account.user && account.user.scopes && account.user.scopes.includes('subscriptions:insights')) return null;
+
+		sendMessage('promoModals.sawInsightsPromo', '', 'metrics');
+
+		return (
+			<InsightsPromoModal
+				show={!this.state.isInsightsModalHidden}
```

Comment:
If we get to this point we should always be rendering, no?

Reply:
Good catch. That variable's used to temporarily disable the insights modal if the user clicks "Sign in" and is redirected to the login page. 

![image](https://user-images.githubusercontent.com/10860129/67300988-ce14a800-f4bc-11e9-90d2-7ed4bd39bc22.png)

I think it would be cleaner to remove this prop entirely and express it as a gate along with the other checks. 

![image](https://user-images.githubusercontent.com/10860129/67300374-d9b39f00-f4bb-11e9-9d9f-2639c0a43fa2.png)